x-pack/plugins/security_solution/server/endpoint/routes/protection_updates_note/handlers.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  RequestHandler,
+  SavedObjectReference,
+  SavedObjectsClientContract,
+} from '@kbn/core/server';
+import type { TypeOf } from '@kbn/config-schema';
+import { PACKAGE_POLICY_SAVED_OBJECT_TYPE } from '@kbn/fleet-plugin/common';
+import { protectionUpdatesNoteSavedObjectType } from '../../lib/protection_updates_note/saved_object_mappings';
+import type {
+  UpdateProtectionUpdatesNoteSchema,
+  GetProtectionUpdatesNoteSchema,
+} from '../../../../common/api/endpoint/protection_updates_note/protection_updates_note_schema';
+
+const findProtectionNote = async (SOClient: SavedObjectsClientContract, policyId: string) => {
+  return SOClient.find<{ note: string }>({
+    type: protectionUpdatesNoteSavedObjectType,
+    hasReference: { type: PACKAGE_POLICY_SAVED_OBJECT_TYPE, id: policyId },
+  });
+};
+
+const updateProtectionNote = async (
+  SOClient: SavedObjectsClientContract,
+  noteId: string,
+  note: string,
+  references: SavedObjectReference[]
+) => {
+  return SOClient.update(
+    protectionUpdatesNoteSavedObjectType,
+    noteId,
+    {
+      note,
+    },
+    {
+      references,
```

Comment:
Not sure about how the `references` attribute works, could this block the pakage policy being removed? Since we remove the note with the `after remove` hook.

Reply:
I think we should ask around. However, I am able to remove the policy without removing the SO with reference.